homeassistant/components/tessie/lock.py

```diff
@@ -81,18 +85,56 @@ def is_locked(self) -> bool | None:
 
     async def async_lock(self, **kwargs: Any) -> None:
         """Enable speed limit with pin."""
+        ir.async_create_issue(
+            self.coordinator.hass,
+            DOMAIN,
+            "deprecated_speed_limit_locked",
+            breaks_in_ha_version="2024.10.0",
+            is_fixable=False,
+            is_persistent=True,
+            severity=ir.IssueSeverity.WARNING,
+            translation_key="deprecated_speed_limit_locked",
+        )
         code: str | None = kwargs.get(ATTR_CODE)
         if code:
             await self.run(enable_speed_limit, pin=code)
             self.set((self.key, True))
 
     async def async_unlock(self, **kwargs: Any) -> None:
         """Disable speed limit with pin."""
+        ir.async_create_issue(
+            self.coordinator.hass,
+            DOMAIN,
+            "deprecated_speed_limit_unlocked",
+            breaks_in_ha_version="2024.10.0",
+            is_fixable=False,
+            is_persistent=True,
+            severity=ir.IssueSeverity.WARNING,
+            translation_key="deprecated_speed_limit_unlocked",
+        )
         code: str | None = kwargs.get(ATTR_CODE)
         if code:
             await self.run(disable_speed_limit, pin=code)
             self.set((self.key, False))
 
+    async def async_added_to_hass(self) -> None:
```

Comment:
Could we avoid creating this entity at all if it's not already existing (in `setup_entry()`)?

Reply:
Yes but then I couldn't figure out how to test it because it will never exist.

Reply:
You would know what the entity_id will be when it would be created so assert that it's not in entity registry.

Reply:
But how do I test the code for when the entity does exist? I did what your asking in a previous commit and it killed the code coverage.

Reply:
Before setting up the integration manually create the entity in entity reg. 

We had this previously for most weather integrations so if you check any of those prior to 2024.4 beta cut there should be examples. I can provide examples tomorrow.

Reply:
Here is one example how it was made for `met` to only create the entity if it was already there + tests. 

https://github.com/home-assistant/core/pull/98098

Reply:
I have commited my attempt at implementing this but I cant get it working due to an odd error.

`TypeError: unhashable type: 'set'`

Reply:
See my new comments to fix that